### PR TITLE
Add Jigsaw trasformation

### PR DESCRIPTION
**Pull request by shikharmn**
Jigsaw transformation returns n_grid random crops from the input image. This augmentation is essential for multiple Self Supervised Learning tasks. Part of a adding PIRL to the framework, as per #396.

### CODE REVIEW

**Comment by IgorSusmelj** [comment]
This looks great thanks! Could you add an example of how to use it?

**Comment by shikharmn** [comment]
Yeah, sure. Where exactly should I add this example?

**Comment by shikharmn** [comment]
I've added an example usage to the jigsaw.py file as I've seen in other parts of the code, do let me know if this works.

File: tests/transforms/test_Jigsaw.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from PIL import Image
+import unittest
+
+from lightly.transforms import Jigsaw
+
+
+class TestGaussianBlur(unittest.TestCase):
```
Comment (IgorSusmelj) [inline tests/transforms/test_Jigsaw.py:7]: rename to `TestJigsaw`

**Review by IgorSusmelj — CHANGES_REQUESTED** [review]
Please rename the class for the test. Otherwise this looks great and we can merge it :)
Thanks a lot!

**Comment by shikharmn** [comment]
Oops, done.

**Review by IgorSusmelj — APPROVED** [review]

**Pull request merged**